sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -320,3 +289,138 @@ def write(self, prefix, tag, m):
             if self.comm is None or self.comm.rank == 0:
                 enmap.write_map(oname, m)
         return oname
+
+class DemodSignalMapHealpix(DemodSignal):
```

Comment:
I don't think it will be sustainable to have a separate class just to support Healpix projection.  Or more to the point: this class and DemodSignalMap contain too much complex, duplicated code.  We have to find a re-factoring that reduces such duplication.

Reply:
@mhasself these are very similar to the Signal class that are in the mapmaker.py. 

That can be the initial abstraction and move from there

Reply:
Ya, that's certainly the origin of DemodSignalmap... but stripped down for simple filter+bin use.  We should indeed be unforking these classes, not forking them more.

Reply:
I agree with Matthew, I don't think we should have a new class.

Reply:
Happy to refactor this. Perhaps a single class with two separate constructors `DemodSignalMap.for_rectpix` and `DemodSignalMap.for_healpix`, following the model of `pmat.P`? The base `__init__` would stay the same but with all args turned into kwargs and expanded with healpix kwargs, so existing code calling it will not break. 
Then I should be able to set a `pix_scheme` flag and navigate the branches straightforwardly as in `pmat`. 

If this sounds acceptable I'll proceed with the implementation, let me know of any other suggestions.

Reply:
That sounds good to me!